### PR TITLE
Add GCP cluster support to OSD provision script

### PR DESCRIPTION
**Pull request by obrienrobert**
[MGDSTRM-8938](https://issues.redhat.com/browse/MGDSTRM-8938)

Update the `osd-provision.sh` script to allow GCP clusters to be provisioned. The script contains sample commands on how to create a GCP cluster. A GCP service account is required. The script will default to `--cloud-provider aws`,  so it shouldn't break any existing CI.

### CODE REVIEW

**Review by kornys — APPROVED** [review]

**Review by MikeEdgar — COMMENTED** [review]
Have you seen this error before? I wonder if it's something to do with my account. The server JSON that the script generates does have the expected `"compute": 3`
```json
{
  "kind": "Error",
  "id": "400",
  "href": "/api/clusters_mgmt/v1/errors/400",
  "code": "CLUSTERS-MGMT-400",
  "reason": "required total number of '48' nodes: '24' master nodes, '12' infra nodes and '12' compute nodes, exceeds the available quota of '24'",
  "operation_id": "d9fe9430-7d15-4cdf-8dca-f043ecc75190"
}
```

**Comment by obrienrobert** [comment]
> Have you seen this error before? I wonder if it's something to do with my account. The server JSON that the script generates does have the expected `"compute": 3`
> 
> ```json
> {
>   "kind": "Error",
>   "id": "400",
>   "href": "/api/clusters_mgmt/v1/errors/400",
>   "code": "CLUSTERS-MGMT-400",
>   "reason": "required total number of '48' nodes: '24' master nodes, '12' infra nodes and '12' compute nodes, exceeds the available quota of '24'",
>   "operation_id": "d9fe9430-7d15-4cdf-8dca-f043ecc75190"
> }
> ```

I haven't, and I didn't come across it when testing. It seems to be quota related, so maybe the quota is exhausted in that account? I'm really not 100% sure.

**Review by franvila — APPROVED** [review]

**Review by MikeEdgar — APPROVED** [review]
Thanks - I get that error using `ocm` directly anyway, so certainly not an issue with the changes.

LGTM

**Comment by rareddy** [comment]
I verified too, this works!

**Comment by obrienrobert** [comment]
@MikeEdgar @rareddy would you be able to merge this if you are happy with it? I don't have merge rights in this repo.

**Pull request merged**
